contrib/lkt/language/prelude.lkt

```diff
@@ -35,7 +35,7 @@ trait Iterator {
 
 @builtin generic[T]
 struct Array implements Sized, Indexable[T], Iterator[T] {
-    @builtin @property fun to_iterator(): Iterator[T]
+    @builtin fun to_iterator(): Iterator[T]
```

Comment:
The commit message is not very informative: why is it a fix? Looking a the change, it looks we just want to change the signature of `to_iterator` (API change, not a bug fix) Also I don’t understand why this is needed even looking at the next commits: please add a testcase and/or expand the commit message.

Reply:
Commit msg has been updated as follows: 

>>> 
Will be needed for an upcoming fix on call resolution. For the moment
to_iterator is unparsed as a regular function call in dsl_unparse, but
in the prelude it's a property, so unify towards it being a function.
Some discrepancies (see entity_iterator test) are ignored for the moment
due to a hole in type resolution.

Reply:
Perfect, thanks!